### ISSUE TITLE
Incorrect type hint for alias of a module which is imported by its submodule

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#4752]
**Describe the bug**
If a module is imported by `import <module>.<submodule>` and then is assigned to an alias, type hint of this alias is incorrect (while its functionality is all good).

**To Reproduce**
```python
import numpy.typing
np = numpy
reveal_type(np)  # Type of np is "Module("numpy.typing")"
np.NDArray  # Flagged. Working as intended.
np.typing.NDArray  # Not flagged. Working as intended.
```

**Expected behavior**
Type hint in the sample should be "Module("numpy")".

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/15389744/223908353-d9f19c08-0687-4652-9b06-59bc58c0bc06.png)

**VS Code extension or command-line**
Both.

Pylance: v2023.3.21 Pre-release
Pyright: 1.1.298

**Comment by erictraut** [comment]
This is just a display issue, but I agree it's a bug.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.299, which I just published. It will also be included in a future release of pylance.
